src/server/middleware/v2auth/auth.go

```diff
@@ -60,7 +60,8 @@ func (rc *reqChecker) check(req *http.Request) (string, error) {
 				return getChallenge(req, al), fmt.Errorf("unauthorized to list catalog")
 			}
 		}
-		if a.target == repository && req.Header.Get(authHeader) == "" && req.Method == http.MethodHead { // make sure 401 is returned for CLI HEAD, see #11271
+		if a.target == repository && req.Header.Get(authHeader) == "" &&
+			(req.Method == http.MethodHead || req.Method == http.MethodGet) { // make sure 401 is returned for CLI HEAD, see #11271
 			return getChallenge(req, al), fmt.Errorf("authorize header needed to send HEAD to repository")
```

Comment:
The error message here says `authorize header needed to send HEAD to repository` but this is true for `GET` too (with this PR), right? With `(req.Method == http.MethodHead || req.Method == http.MethodGet)`

cc @reasonerjt

Reply:
Should it says `HEAD or GET` ? Or maybe interpolate the `req.Method` in the string for accuracy? Something like this maybe? -

```go
fmt.Errorf("authorize header needed to send %s to repository", req.Method)
```